File: Material/Codigo/Frontend/src/app/interfaces/reservation.ts
export interface ReservationRequest {
    pharmacyId: number;
    userEmail: string;
    details: ReservationDetailRequest[];
    prescriptionBase64?: string;
    prescriptionFileName?: string;
}

export interface ReservationDetailRequest {
    drugCode: string;
    quantity: number;
}

export interface ReservationResponse {
    id: number;
    code: string;
    publicKey: string;
    pharmacyId: number;
    userEmail: string;
    reservationDate: string;
    expirationDate: string;
    status: string;
    details: ReservationDetailResponse[];
    prescriptionUploaded?: boolean;
    hasRecipe: boolean;
    requiresPrescription: boolean;
    recipeFiles: string[];
}

export interface ReservationDetailResponse {
    id: number;
    drugCode: string;
    drugName?: string;
    quantity: number;
    requiresPrescription: boolean;
}

File: Material/Codigo/Backend/PharmaGo.Domain/Entities/Reservation.cs
using PharmaGo.Domain.Enums;

namespace PharmaGo.Domain.Entities
{
    public class Reservation
    {
        public int Id { get; set; }
        public string Code { get; set; }
        public string PublicKey { get; set; }
        public string PrivateKey { get; set; }
        public ICollection<ReservationDetail> Details { get; set; }
        public int PharmacyId { get; set; }
        public string UserEmail { get; set; }
        public DateTime ReservationDate { get; set; }
        public ReservationStatus Status { get; set; }
        public string? PrescriptionBase64 { get; set; }
        public string? PrescriptionFileName { get; set; }
    }
}
